animations/lib/src/basics/08_animated_list.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class AnimatedListDemo extends StatefulWidget {
+  static String routeName = '/basics/08_animated_list';
+
+  _AnimatedListDemoState createState() => _AnimatedListDemoState();
+}
+
+class _AnimatedListDemoState extends State<AnimatedListDemo> {
+  final GlobalKey<AnimatedListState> _listKey = GlobalKey();
+
+  void addUser() {
+    int index = listData.length;
+    listData.add(listData[Random().nextInt(5)]);
+    _listKey.currentState
+        .insertItem(index, duration: Duration(milliseconds: 500));
```

Comment:
While you've done a good job customizing the animations, I'd recommend dropping this duration change. The default is 300ms, and any complexity that can be pulled out of a sample helps the people trying to learn from it.